### DISCUSSION TITLE
Return folders only from a certain subfolder

### DISCUSSION THREAD

**Discussion by gitjmartin** [microsoft-search/pnp-modern-search#3181] [Q&A] [answered]
We have a document library that has a number of root folders in it with files and folders under each.  I'd like to anchor my results to show only those first level subfolders UNDER a particular root folder...if that makes sense.  In the image below, I'd like to anchor to RootFolderC and display all the green highlighted folders (no files and no sub-folders or files UNDER the green highlighted folders).  Any thoughts?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/141683133/61bf5d59-e986-4dd0-b48a-903860bef366)

**Comment by wobba** [comment, marked as answer]
If you have library you can get only the sub-level folder like this:

`path:https://tenant.sharepoint.com/sites/asite/doclib UrlDepth:5 ContentTypeId:0x0120*`

* ContentTypeId:0x0120* limits to folders
* path limits to a document library
* UrlDepth is URL split on /, and 5 would be the second level in a document library

**Reply by gitjmartin** [reply]
Brilliant!  Thanks!

**Comment by MartinM85** [comment]
Is there any way to exclude OneNote files?
`ContentTypeId:0x0120*` doesn't exclude OneNote files.

**Reply by kasperbolarsen** [reply]
I can't remember the file extention for a One Note file but I guess you should be able to exclude those by using the extention?

**Reply by wobba** [reply]
Add `NOT ProgID=OneNote.Notebook` to exclude OneNote, or exclude SiteAssets by either scoping results to a specific library or add `NOT ParentLink:SiteAssets` to exclude any result from the SiteAssets library.

Of course... OneDrive OneNotes are stored different so you need to use ProgID.

**Reply by MartinM85** [reply]
Great, `NOT ProgID=OneNote.Notebook` works. 
Thank you

**Comment by sympmarc** [comment]
This is a great tip! I just made a PowerShell script I'm using much more efficient with `UrlDepth`. Thanks for sharing, @wobba.
